nmpolicy/internal/ast/types.go

```diff
@@ -0,0 +1,16 @@
+package ast
+
+type Meta struct {
+	Position int `yaml:"pos"`
+}
+
+type TernaryOperator [3]Node
+type VariadicOperator []Node
+
+type Node struct {
+	Meta     `yaml:",inline"`
```

Comment:
I think that with the standard json parsing, annotating the embedded structures was not needed. Just noting.